widgets/DownloadIcsLink.php

```diff
@@ -28,9 +24,11 @@ class DownloadIcsLink extends Widget
     public function run()
     {
         if ($this->calendarEntry === null) {
-            return;
+            return '';
         }
 
-        return Html::a(Yii::t('CalendarModule.base', 'Download ICS'), Url::toEntryDownloadICS($this->calendarEntry), ['target' => '_blank']);
+        return $this->render('downloadLink', [
```

Comment:
```suggestion
        return Html::tag(
            'span',
            Html::a(
                Yii::t('CalendarModule.base', 'Download ICS'),
                Url::toEntryDownloadICS($this->calendarEntry),
                ['target' => '_blank'],
            ),
            [
                'id' => 'ics-download',
            ],
        );
```

Reply:
@luke- Yes, I agree, more simple is best, even if we won't be able to overwrite the view, be we can still use Widget events.

We just need to use `class` instead of `id` because there will be several in the same page (in the Stream page).
I'll send a new commit.

Just to make sure before: should the class be `ics-download` or prefixed with the name of the module such as:
-  `calendar-ics-download`
-  `calendar-entry-ics-download`
-  `calendar-entry-ics-download-link`

Reply:
@luke- thanks, done.